zstd/_generate/gen.go

```diff
@@ -770,57 +1195,20 @@ func (e executeSimple) generateProcedure(name string) {
 
 		Comment("Copy non-overlapping match")
 		{
-			e.copyMemory("2", src, outBase, ml)
-			ADDQ(ml, outBase)
-			ADDQ(ml, outPosition)
+			e.copyMemory("2", src, c.outBase, ml)
```

Comment:
This fixes the crash:
```suggestion
			e.copyMemoryPrecise("2", src, c.outBase, ml)
```

We can merge with this and optimize later.

We can do a version with precise copying when we don't have enough overhead.

Reply:
It would be perfect if we can tell upfront which version can be used. We may add some runtime checks before copying.